src/mapgen.cpp

```diff
@@ -589,13 +589,13 @@ class jmapgen_alternativly : public jmapgen_piece {
     // PieceType, they *can not* be of any other type.
     std::vector<PieceType> alternatives;
     jmapgen_alternativly() = default;
-    void apply( map &m, const size_t x, const size_t y, const float mon_density ) const override
+    void apply( map &m, const size_t x, const size_t y, const size_t xmin, const size_t ymin, const size_t xmax, const size_t ymax, const float mon_density ) const override
```

Comment:
Why add both `xmin` and `xmax`? That gives 3 coords just for the x alone. I'm pretty sure `xmin` could be removed and `x` used in its place. Same for `ymin/ymax/y`.